### PR TITLE
Proposed template for bug reporting

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
Part of https://github.com/narwhals-dev/narwhals/issues/218
 
After your (kind 😇) feedback , I'll work on docs improvement and feature request templates... if y'all want. : ) 

Here is how it looks on a mock project:

[New Issue.pdf](https://github.com/narwhals-dev/narwhals/files/15485281/New.Issue.pdf)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- type: textarea
+  attributes:
+    label: Steps or code to reproduce the bug
+    description: |
+       Please add a minimal code example to reproduce the bug.
+  validations:
+       required: true 
+- type: textarea
+  attributes:
+    label: Expected Results
```
Comment (InessaPawson) [inline .github/ISSUE_TEMPLATE/bug_report.yml:32]:
```suggestion
    label: Expected results
```
Let's keep the formatting of the heading consistent with the rest of the template.

**Review by InessaPawson — COMMENTED** [review]
@DeaMariaLeon I left 2 minor suggestions. The template looks great!

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    description: >
+       Please paste or describe the expected results.
+    placeholder: >
+        Example: No error is thrown.
+  validations:
+        required: true
+- type: textarea
+  attributes:
+    label: Actual Results
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Thanks Inessa. It's done.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Same. :-)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    label: Actual results
+    description: |
+      Please paste or describe the results you observe instead of the expected results. 
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Versions
+    description: >
+      What version of Narwhals are you running?
```
Comment (FBruzzesi) [inline .github/ISSUE_TEMPLATE/bug_report.yml:50]: I would consider asking for pandas/polars or other dataframe library version as well.
Maybe even suggest how to find those? A naive way would probably be something like:

```terminal
pip freeze | grep -E "narwhals|polars|pandas|modin|cudf"
```

**Review by FBruzzesi — COMMENTED** [review]
That is an amazing template! I didn't even know it was possible 🙌🏼

**Review by MarcoGorelli — COMMENTED** [review]
nice one, thanks! looks close, let's get this in

separately, we also probably need a "feature request" template?

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: I added the suggestion. If it still needs change, let me know please.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+name: Bug report
+description: Report to help us reproduce the bug
+labels: ['Needs Triage']
+title: "[Bug]: "
+
+body:
+- type: markdown 
+  attributes:
+    value: >
+        ## Thanks for taking the time to fill out this report
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml:9]: I took it from github's documentation (for templates).

**Comment by DeaMariaLeon** [comment]
> separately, we also probably need a "feature request" template?

Yes this is planned. 

I think I added all the feedback. Feel free to change/add more stuff.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      Please paste or describe the results you observe instead of the expected results. 
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Versions
+    render: shell
+    description: |
+      What version of Narwhals and dataframe libraries are you running?
+      You could run the following and provide the output bellow.
```
Comment (MarcoGorelli) [inline .github/ISSUE_TEMPLATE/bug_report.yml:51]:
```suggestion
      You could run the following and provide the output below.
```

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks @DeaMariaLeon

**Pull request merged**